src/Sentry/Hint.cs

```diff
@@ -0,0 +1,88 @@
+namespace Sentry;
+
+/// <summary>
+/// A hint that can be provided when capturing a <see cref="SentryEvent"/> or adding a <see cref="Breadcrumb"/>.
+/// Hints can be used to filter or modify events or breadcrumbs before they are sent to Sentry.
+/// </summary>
+public class Hint
+{
+    private readonly List<Attachment> _attachments = new();
+    private readonly Dictionary<string, object?> _items = new();
+
+    /// <summary>
+    /// Creates a new instance of <see cref="Hint"/>.
+    /// </summary>
+    public Hint()
+    {
+    }
+
+    /// <summary>
+    /// Creates a new hint with a single key/value pair.
+    /// </summary>
+    /// <param name="key"></param>
+    /// <param name="value"></param>
+    public Hint(string key, object? value)
+        : this()
+    {
+        _items[key] = value;
+    }
+
+    internal void AddAttachmentsInternal(IEnumerable<Attachment> attachments)
+    {
+        if (attachments is not null)
+        {
+            _attachments.AddRange(attachments);
+        }
+    }
+
+    /// <summary>
+    /// Adds one or more attachments to the Hint.
+    /// </summary>
+    /// <param name="attachments"></param>
+    public void AddAttachments(params Attachment[] attachments) => AddAttachmentsInternal(attachments);
+
+    /// <summary>
+    /// Adds multiple attachments to the Hint.
+    /// </summary>
+    /// <param name="attachments"></param>
+    public void AddAttachments(IEnumerable<Attachment> attachments) => AddAttachmentsInternal(attachments);
+
+    /// <summary>
+    /// Attachments added to the Hint.
+    /// </summary>
+    public ICollection<Attachment> Attachments => _attachments;
+
+    /// <summary>
+    /// Data provided with the Hint.
+    /// </summary>
+    public IDictionary<string, object?> Items => _items;
+
+    /// <summary>
+    /// Gets or sets a Screenshot for the Hint
+    /// </summary>
+    public Attachment? Screenshot { get; set; }
+
+    /// <summary>
+    /// Gets or sets a ViewHierarchy for the Hint
+    /// </summary>
+    public Attachment? ViewHierarchy { get; set; }
+
```

Comment:
Let's remove these properties.  Reasons:

- They aren't part of the `Attachments` collection.
- They assume there can only be one screenshot or view hierarchy - but there could be multiple (multiple windows, multi-screen devices, etc.)
- We only have limited support for view hierarchy in the base sdk.  Presently only `sentry-unity` uses it.
- We don't have any support for screenshots in the base sdk yet.

Reply:
Done.